vdr/didsubject/did_document.go

```diff
@@ -0,0 +1,189 @@
+/*
+ * Copyright (C) 2024 Nuts community
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+package didsubject
+
+import (
+	"encoding/json"
+	"errors"
+	"github.com/google/uuid"
+	ssi "github.com/nuts-foundation/go-did"
+	"github.com/nuts-foundation/go-did/did"
+	"github.com/nuts-foundation/nuts-node/jsonld"
+	"gorm.io/gorm"
+	"gorm.io/gorm/schema"
+	"time"
+)
+
+// DIDDocument is the gorm representation of the did_document_version table
+type DIDDocument struct {
+	ID        string `gorm:"primaryKey"`
+	DidID     string `gorm:"column:did"`
+	DID       DID    `gorm:"foreignKey:DidID;references:ID"`
+	CreatedAt int64  `gorm:"autoCreateTime:false"`
+	// UpdatedAt is the (unix) timestamp when the document was updated
+	// Also used to purge DID document changes that haven't been committed within a certain time frame
+	UpdatedAt           int64 `gorm:"autoUpdateTime:false"`
+	Version             int
```

Comment:
both are strictly increasing functions to determine the ordering of documents. Do we need both?

Reply:
updatedAt could be the same for two versions. It's only seconds